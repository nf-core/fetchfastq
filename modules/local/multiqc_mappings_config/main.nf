
process MULTIQC_MAPPINGS_CONFIG {

    conda "conda-forge::python=3.9.5"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/python:3.9--1' :
        'biocontainers/python:3.9--1' }"

    input:
    Path csv

    output:
    Path yml = path("multiqc_config.yml")

    topic:
    [ task.process, 'python', eval("python --version | sed 's/Python //g'") ] >> 'versions'

    script:
    """
    multiqc_mappings_config.py \\
        $csv \\
        multiqc_config.yml
    """
}
