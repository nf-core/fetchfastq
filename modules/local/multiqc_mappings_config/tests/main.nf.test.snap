{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "multiqc_config.yml:md5,7f3cb10fff83ba9eb3e8fa6862d1290a"
                ],
                "1": [
                    "versions.yml:md5,dd4c66f0551d15510b36bb2e2b2fdd73"
                ],
                "versions": [
                    "versions.yml:md5,dd4c66f0551d15510b36bb2e2b2fdd73"
                ],
                "yml": [
                    "multiqc_config.yml:md5,7f3cb10fff83ba9eb3e8fa6862d1290a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-17T09:16:46.557614"
    }
}